Options/Applicative/Types.hs

```diff
@@ -319,12 +319,29 @@ type Args = [String]
 
 -- | Policy for how to handle options within the parse
 data ArgPolicy
-  = SkipOpts  -- ^ Inputs beginning with `-` or `--` are treated
-              --   as options or flags, and can be mixed with arguments.
-  | AllowOpts -- ^ All input is treated as positional arguments.
-              --   Used after a bare `--` input, and also with
-              --   `noIntersperse` policy.
-  deriving (Eq, Show)
+  = InterspersePolicy
+  -- ^ The default policy, options and arguments can
+  --   be interspersed.
+  --   A `--` option can be passed to ensure all following
+  --   commands are treated as arguments.
+  | NoInterspersePolicy
+  -- ^ Options must all come before arguments, once a
+  --   single option is parsed, all remaining arguments
+  --   are treated as positionals.
```

Comment:
`s/once a single option is parsed/once a single argument is parsed/` ? (unless I misunderstand the behaviour)